backend/kesaseteli/applications/api/v1/views.py

```diff
@@ -229,10 +256,33 @@ def activate(self, request, *args, **kwargs) -> HttpResponse:
         list(same_persons_apps)  # Force evaluation of queryset to lock its rows
 
         if same_persons_apps.active().exists():
-            return HttpResponseRedirect(youth_application.already_activated_page_url())
+            if (
+                youth_application.need_additional_info
+                and not youth_application.has_additional_info
+            ):
+                return HttpResponseRedirect(
+                    youth_application.additional_info_page_url(pk=youth_application.pk)
+                )
+            else:  # Additional info is not needed or it has been provided
```

Comment:
it could also redirect here to additional info page because the page would just show the "additional info application sent succesfully" -message